anvio/threadingops.py

```diff
@@ -0,0 +1,551 @@
+        """Checks that `peptide_paths` files actually exist, then combines them."""
+        # For renaming fasta headers
+        hit_id = 0
+
+        # Set up data storage.
+        # Todo: These are keyed with ints, so idk why they aren't lists.  Maybe used as a dict later in the code?
+        gene_calls_dict = {}  # each entry must contain {'contig', 'start', stop, 'direction', 'partial'} items.
+        amino_acid_sequences_dict = {}
+
+        for peptide_path in peptide_paths:
+            if not os.path.exists(peptide_path):
+                if self.logger:
+                    self.logger.progress.end()
+
+                raise ConfigError("Something went wrong with prodigal, and it failed to generate the "
+                                  "expected output ('%s') :/ Fortunately, this log file should tell you what "
+                                  "might be the problem: '%s'. Please do not forget to include this "
+                                  "file if you were to ask for help." % (peptide_path, self.log_file_path))
+
+            # Some splits may not actually have gene calls.  Skip them.
+            if filesnpaths.is_file_empty(peptide_path):
+                continue
+
+            # If we get here, the fasta file will not be empty.
+            fasta = fastalib.SequenceSource(peptide_path)
+
+            while next(fasta):
+                gene_calls_dict[hit_id] = self.parser(fasta.id)
+                amino_acid_sequences_dict[hit_id] = fasta.seq.replace('*', '')
+                hit_id += 1
+
+            fasta.close()
+
+            # todo i think this is removed elsewhere
+            # Remove the split peptide file.
+            os.remove(peptide_path)
+
+        # If no genes were predicted across all output files, warn the user.
+        if len(amino_acid_sequences_dict) == 0:
+            if self.logger:
+                self.logger.run.info('Result',
+                                     f'Prodigal ({self.installed_version}) has identified no genes :/',
+                                     nl_after=1,
+                                     mc="red")
+        else:  # Write out the final gene file
+            assert 'peptide_path' in self.collated_output_file_paths
+            with open(self.collated_output_file_paths['peptide_path'], 'w') as f:
+                for hit_id, sequence in amino_acid_sequences_dict.items():
+                    f.write(f">{hit_id}\n{sequence}\n")
+
+        return gene_calls_dict, amino_acid_sequences_dict
+
+    # Check the gff files and append them into the single outfile.
+    def _process_gff_files(self, gff_paths: Collection[str]) -> None:
+        """Check the gff files and combine them.
+
+        Question:  Is genes_in_contigs (the gff file) ever actually used?...each of the threads will start renumbering
+        sequences from 1 in each subfile.  Will this break something downstream?
+        """
```

Comment:
No, GFF files are not used for anything (if we did, numbering that reset in each file would have been a big problem though :)).